docs/source/user_guide/basic.rst

```diff
@@ -180,34 +181,39 @@ We can also use :meth:`cupy.ndarray.get()`:
 Memory management
 -----------------
 
-Check :doc:`./memory` for a detailed description of how is memory managed in CuPy
+Check :doc:`./memory` for a detailed description of how memory is managed in CuPy
 using memory pools.
 
 
-How to write CPU/GPU agnostic code
+How to write device agnostic code
 ----------------------------------
 
-The compatibility of CuPy with NumPy enables us to write CPU/GPU generic code.
-It can be made easy by the :func:`cupy.get_array_module` function.
-This function returns the :mod:`numpy` or :mod:`cupy` module based on arguments.
-A CPU/GPU generic function is defined using it like follows:
+CuPy's compatibility with NumPy makes it possible to write device agnostic code.
+For this purpose, CuPy implements the :func:`cupy.get_array_module` function, that
+returns a reference to :mod:`cupy` if any of its arguments resides on a GPU
+and :mod:`numpy` otherwise.
+Here is an example of a device agnostic function that computes ``log1p``:
 
 .. doctest::
 
    >>> # Stable implementation of log(1 + exp(x))
    >>> def softplus(x):
-   ...     xp = cp.get_array_module(x)
-   ...     return xp.maximum(0, x) + xp.log1p(xp.exp(-abs(x)))
+   ...     mod = cp.get_array_module(x)
+   ...     print("Using module:", mod.__name__)
+   ...     return mod.maximum(0, x) + mod.log1p(mod.exp(-abs(x)))
```

Comment:
Please revert this example. `xp` is a standard usage that already prevails in the community.

Reply:
I'm OK with reverting this, but wanted to note that this document targets those who are not part of the community yet. So, do you think a note about `xp` (in the form of a Python comment) would fit this example?

Reply:
> this document targets those who are not part of the community yet

I wouldn't say so. The current documentation is written under an (implicit) assumption that users are already familiar with NumPy (so `import numpy as np` shouldn't be considered foreign) and basic terminology of GPU programming (ex: users need to know the distinction between host and device, data locality, etc). This, compounded with `import cupy as cp`, renders `xp` here an easy-to-understand convention.   

IIRC there is a plan to write a new piece of docs targeting beginners, just like NumPy's (https://numpy.org/doc/stable/user/absolute_beginners.html). We could explain this convention there. (Maybe you're interested in contributing to this write-up? 😇)

Reply:
> > this document targets those who are not part of the community yet
> 
> I wouldn't say so. The current documentation is written under an (implicit) assumption that users are already familiar with NumPy (so `import numpy as np` shouldn't be considered foreign) and basic terminology of GPU programming (ex: users need to know the distinction between host and device, data locality, etc). This, compounded with `import cupy as cp`, renders `xp` here an easy-to-understand convention.

I'm sorry I wasn't clear -- I was talking about the CuPy community, not NumPy/GPU programming beginners: this is the "Basics" page of the CuPy's user's guide, so the target audience is those who are unfamiliar with CuPy. For example, I know about `numpy/np` convention and this page introduces `cupy/cp` but `xp` didn't strike me as super obvious. I think once the reader's knowledge of `cupy/cp` convention solidifies, `xp` var. name will make more sense.